src/export/StructureDefinitionExporter.ts

```diff
@@ -689,14 +689,22 @@ export class StructureDefinitionExporter implements Fishable {
     }
     rule.items.forEach(item => {
       if (item.type) {
-        const extension = this.fishForFHIR(item.type, Type.Extension);
+        // there might be a |version appended to the type, so don't include it while fishing
+        const [typeWithoutVersion, version] = item.type.split('|', 2);
+        const extension = this.fishForFHIR(typeWithoutVersion, Type.Extension);
         if (extension == null) {
           logger.error(
             `Cannot create ${item.name} extension; unable to locate extension definition for: ${item.type}.`,
             rule.sourceInfo
           );
           return;
         }
+        if (version != null && extension.version != null && version != extension.version) {
+          logger.warn(
+            `${item.type} is based on ${typeWithoutVersion} version ${version}, but SUSHI found version ${extension.version}`,
```

Comment:
For some reason the phrase *based on* just doesn't sound right to me. Might it be more clear if we said:
> The ${typeWithoutVersion} extension was specified with version ${version}, but SUSHI found and used version ${extension.version}

Or:

> The extension ${item.type} was specified, but SUSHI found and used ${typeWithoutVersion}|${extension.version}

Reply:
I'm not sure if this factors in at all, but it is worth noting that this is the same wording we have in the two other places we have a similar message in ElementDefinition [here](https://github.com/FHIR/sushi/blob/master/src/fhirtypes/ElementDefinition.ts#L1245-L1249) and [here](https://github.com/FHIR/sushi/blob/master/src/fhirtypes/ElementDefinition.ts#L2325-L2329).

Reply:
Ah. I see. I should have guessed! The other ones are a little different because they're dealing w/ canonicals we pulled out of definitions (either `SD.baseDefinition` when traversing the hierarchy or `ED.type.profile` when unfolding); whereas this one is a canonical version explicitly specified by the author.  The "based on" wording feels a little weird to me in all of them -- but I don't think they necessarily should all get the same fixed up language.

I'd still suggest we update the message here, but leave it to Mint to decide if they want to update the other messages as well (as they're clearly outside the scope of this PR).